packages/wallet/backend/src/paymentPointer/service.ts

```diff
@@ -64,107 +81,175 @@ export const createPaymentPointerIfFalsy = async ({
+        account.userId === args.userId
       ) {
         paymentPointer = await PaymentPointer.query().patchAndFetchById(
           paymentPointer.id,
           {
-            publicName,
+            publicName: args.publicName,
             active: true
           }
         )
       }
     } else {
       const rafikiPaymentPointer =
         await this.deps.rafikiClient.createRafikiPaymentPointer(
-          publicName,
+          args.publicName,
           account.assetId,
           url
         )
 
+      let assetScale = null
+      let assetCode = null
+      if (args.isWM) {
+        //* Web monetization feature requires an asset with scale MAX_ASSET_SCALE to exist. It's default assetCode is USD for now
+        const webMonetizationAsset =
+          await this.deps.rafikiClient.getRafikiAsset(
+            'USD',
+            this.deps.env.MAX_ASSET_SCALE
+          )
+
+        if (!webMonetizationAsset) {
+          throw new NotFound('Web monetization asset not found')
+        }
+
+        assetScale = webMonetizationAsset.scale
+        assetCode = webMonetizationAsset.code
+      }
+
       paymentPointer = await PaymentPointer.query().insert({
         url: rafikiPaymentPointer.url,
-        publicName,
-        accountId,
-        id: rafikiPaymentPointer.id
+        publicName: args.publicName,
+        accountId: args.accountId,
+        id: rafikiPaymentPointer.id,
+        assetCode,
+        assetScale,
+        balance: 0,
+        isWM: args.isWM
       })
+
+      args.isWM &&
+        (await this.deps.cache.set(paymentPointer.id, paymentPointer, {
+          expiry: 60
+        }))
     }
 
     return paymentPointer
   }
 
-  async list(userId: string, accountId: string): Promise<PaymentPointer[]> {
-    // Validate that account id belongs to current user
+  async list(userId: string, accountId: string): Promise<PaymentPointerList> {
     const account = await this.deps.accountService.findAccountById(
       accountId,
       userId
     )
 
-    return PaymentPointer.query()
+    const paymentPointersResult = await PaymentPointer.query()
       .where('accountId', account.id)
       .where('active', true)
+
+    const result = paymentPointersResult.reduce(
+      (acc, pp) => {
+        if (pp.isWM) {
+          acc.wmPaymentPointers.push(pp)
+        } else {
+          acc.paymentPointers.push(pp)
+        }
+        return acc
+      },
+      {
+        wmPaymentPointers: [] as PaymentPointer[],
+        paymentPointers: [] as PaymentPointer[]
+      }
+    )
+
+    return result
   }
 
   async listAll(userId: string): Promise<PaymentPointer[]> {
-    return PaymentPointer.query().joinRelated('account').where({
-      'account.userId': userId,
-      'paymentPointers.active': true
-    })
+    return PaymentPointer.query()
+      .where({ isWM: false })
+      .joinRelated('account')
+      .where({
+        'account.userId': userId,
+        'paymentPointers.active': true
```

Comment:
should this check be moved above as well?

Reply:
like this?

`return PaymentPointer.query()
      .where({ isWM: false, active: true })
      .joinRelated('account')
      .where({
        'account.userId': userId,
      })`

Reply:
yes